src/aggregator/client/writer.go

```diff
@@ -200,15 +215,9 @@ func (w *writer) encodeWithLock(
 		return err
 	}
 
-	if encoder.Len() < w.maxBatchSize {
-		encoder.Unlock()
-		return nil
-	}
-
-	buffer := encoder.Relinquish()
 	encoder.Unlock()
```

Comment:
Removing batching here makes _current_ queue management policy pretty much useless in the `Queue`. 

Right now queue will drop oldest buffers based on buffer _count_ limit, with this change there's only going to be up to $numShardsPerInstance buffers, so the limit efficiency will be vastly decreased if buffer size is now suddenly an order of magnitude (or two) larger.

You need to change the limit to be total buffer _size_ based and not buffer _count_ in order to keep some safety guarantees/memory bounds.